File: src/main/java/net/rkoubou/kspcompiler/analyzer/PreProcessorSymbol.java
/* =========================================================================

    PreProcessorSymbol.java
    Copyright (c) R-Koubou

   ======================================================================== */

package net.rkoubou.kspcompiler.analyzer;

import net.rkoubou.kspcompiler.javacc.generated.ASTPreProcessorDefine;

/**
 * プリプロセッサで定義したシンボルの中間表現を示す
 */
public class PreProcessorSymbol extends SymbolDefinition
{

    /** 元となるASTノード */
    public final ASTPreProcessorDefine astNode;

    /**
     * Ctor.
     */
    public PreProcessorSymbol( ASTPreProcessorDefine node )
    {
        copy( node.symbol, this );
        this.astNode = node;
        this.symbolType = SymbolType.PreprocessorSymbol;
    }
}

File: src/main/kotlin/net/rkoubou/kspcompiler/analyzer/PreProcessorSymbol.kt
/* =========================================================================

    PreProcessorSymbol.java
    Copyright (c) R-Koubou

   ======================================================================== */

package net.rkoubou.kspcompiler.analyzer

import net.rkoubou.kspcompiler.javacc.generated.ASTPreProcessorDefine
import net.rkoubou.kspcompiler.analyzer.SymbolDefinition

/**
 * プリプロセッサで定義したシンボルの中間表現を示す
 */
class PreProcessorSymbol( val astNode : ASTPreProcessorDefine ) : SymbolDefinition()
{
    init
    {
        SymbolDefinition.copy(astNode.symbol, this)
        this.symbolType = SymbolDefinition.SymbolType.PreprocessorSymbol
    }
}
